File: solutions/2356-largest-combination-with-bitwise-and-greater-than-zero/largest-combination-with-bitwise-and-greater-than-zero.js
/**
 * @param {number[]} candidates
 * @return {number}
 */
var largestCombination = function(candidates) {
    const BitMax = 24;
    let maxSet = 1;
    let mask = 1;
    
    for (let shift = 0; shift < BitMax; shift++) {
        let count = 0;
        for (let num of candidates) {
            if (num & mask) {
                count++;
            }
        }
        maxSet = Math.max(maxSet, count);
        mask <<= 1;
    }
    
    return maxSet;
};

File: solutions/2356-largest-combination-with-bitwise-and-greater-than-zero/largest-combination-with-bitwise-and-greater-than-zero.java
public class Solution {
    private static final int BIT_MAX = 24;
    
    public int largestCombination(int[] candidates) {
        int maxSet = 1;
        int mask = 1;
        
        for (int shift = 0; shift < BIT_MAX; shift++) {
            int count = 0;
            for (int num : candidates) {
                if ((num & mask) != 0) {
                    count++;
                }
            }
            maxSet = Math.max(maxSet, count);
            mask <<= 1;
        }
        return maxSet;
    }
}